### PR TITLE
API RedHat & PyTest & CronJob

### LINKED ISSUE #69: CronTab for WebScraping APIs

### LINKED ISSUE #95: API RedHat Jobs

### PR DESCRIPTION
**Pull request by RomTamam**
Fix #95 
**Fix #69 - WIP**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -5,15 +5,14 @@ verify_ssl = true
 
 [dev-packages]
 flake8 = "*"
-pytest = "*"
```
Comment (simzacks) [inline Pipfile:8]: Why are you removing pytest?

File: cron.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+def scheduled_job():
+    print('CronJob started')
+    f= open('/vagrant/ultimatejob/tests/API_FB.py','w')
+    f.close()
```
Comment (simzacks) [inline cron.py:4]: What are you trying to accomplish by opening the file in write mode and then closing it?

File: tests/API_FB.py [hunk 3]
```diff
@@ -44,6 +44,7 @@ def create_req():
     # Load our first page
     # Need to pull from db
     search_url = "https://www.facebook.com/careers/jobs/?q="
-    prefrences = "DevOps"
-    req = requests.get(search_url+prefrences)
+    company = "Facebook"
+    # prefrences = "DevOps"
+    req = requests.get(search_url)
```
Comment (simzacks) [inline tests/API_FB.py:47]: neither `company` not `prefrences` (commented out) are used in the function.
If they're not needed, remove them

File: tests/API_RH.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Import and load necessary lib
+from bs4 import BeautifulSoup
```
Comment (simzacks) [inline tests/API_RH.py:2]: This is a copy of API_FB. Is there a reason to commit it before modifying it to work for RH?

File: tests/tests_API_FB.py [hunk 5]
```diff
@@ -9,21 +9,61 @@ def test_api_urls():
...
     # Create soup object & print a status message - 404 not good , 200 good
     soup = API_FB.create_soup(API_FB.create_req())
-    expected_output_urls = ['https://www.facebook.com/careers/v2/jobs/1672813472870915/',
-                            'https://www.facebook.com/careers/v2/jobs/712878282607325/',
-                            'https://www.facebook.com/careers/v2/jobs/420212419368641/',
-                            'https://www.facebook.com/careers/v2/jobs/207984144060781/',
-                            'https://www.facebook.com/careers/v2/jobs/773785633436218/',
-                            'https://www.facebook.com/careers/v2/jobs/2597607417128369/',
-                            'https://www.facebook.com/careers/v2/jobs/536078547072736/']
+    expected_output_urls = ['https://www.facebook.com/careers/v2/jobs/1035151026857199/',
```
Comment (simzacks) [inline tests/tests_API_FB.py:12]: Are you sure that they will always be returned in the exact same order?

**Review by simzacks — CHANGES_REQUESTED** [review]
Did you purposefully delete all of those html files?

File: Pipfile (hunk 1 above)
Comment (RomTamam) [reply Pipfile:8]: We're using django-pytest.. this is not the right one
As for your previous question: "Did you purposefully delete all of those html files?" - yes, it's not relevant anymore

File: tests/API_FB.py (hunk 3 above)
Comment (RomTamam) [reply tests/API_FB.py:47]: will do

File: tests/API_RH.py (hunk 4 above)
Comment (RomTamam) [reply tests/API_RH.py:2]: Sorry I don't have permission to define this PR as a draft.. It fixed and will be uploaded in next hour

File: cron.py (hunk 2 above)
Comment (RomTamam) [reply cron.py:4]: Actually I'm still trying to understand how Cron works.. this is supposed to be draft PR

**Comment by RomTamam** [comment]
> 
> 
> Did you purposefully delete all of those html files?

I did, those are our old files..

File: cron.py (hunk 2 above)
Comment (simzacks) [reply cron.py:4]: I marked the PR as draft so it doesn't get accidentally merged

File: tests/API_Amazon.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Import and load necessary lib
+from bs4 import BeautifulSoup
+import requests
+
+
+def extract_jobs_url(soup):
+    fb_url = "https://www.facebook.com"
```
Comment (simzacks) [inline tests/API_Amazon.py:7]: still a copy of FB_API, not converted to amazon yet

File: tests/API_RH.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import requests
+
```
Comment (simzacks) [inline tests/API_RH.py:4]: can you verify that all the functions in this page work?

File: tests/API_Amazon.py (hunk 6 above)
Comment (RomTamam) [reply tests/API_Amazon.py:7]: Last time we talked about the API_RH - it's working now, Amazon still in progress

File: tests/API_RH.py (hunk 7 above)
Comment (RomTamam) [reply tests/API_RH.py:4]: I did, Also pytest works fine

File: tests/tests_API_FB.py (hunk 5 above)
Comment (RomTamam) [reply tests/tests_API_FB.py:12]: Yep, untill Facebook employee will change the list :)
Still looking for a solution for this

**Pull request closed without merging**